src/profile-logic/data-structures.js

```diff
@@ -0,0 +1,264 @@
+    // If modifying this structure, please update all callers of this function to ensure
+    // that they are pushing on correctly to the data structure. These pushes may not
+    // be caught by the type system.
+    address: funcTable.address.slice(),
+    isJS: funcTable.isJS.slice(),
+    relevantForJS: funcTable.relevantForJS.slice(),
+    name: funcTable.name.slice(),
+    resource: funcTable.resource.slice(),
+    fileName: funcTable.fileName.slice(),
+    lineNumber: funcTable.lineNumber.slice(),
+    columnNumber: funcTable.columnNumber.slice(),
+    length: funcTable.length,
+  };
+}
+
+export function getEmptyResourceTable(): ResourceTable {
+  return {
+    // Important!
+    // If modifying this structure, please update all callers of this function to ensure
+    // that they are pushing on correctly to the data structure. These pushes may not
+    // be caught by the type system.
+    lib: [],
+    name: [],
+    host: [],
+    type: [],
+    length: 0,
+  };
+}
+
+export function getEmptyRawMarkerTable(): RawMarkerTable {
+  // Important!
+  // If modifying this structure, please update all callers of this function to ensure
+  // that they are pushing on correctly to the data structure. These pushes may not
+  // be caught by the type system.
+  return {
+    data: [],
+    name: [],
+    time: [],
+    length: 0,
+  };
+}
+
+export function getResourceTypes(): * {
+  return {
+    unknown: 0,
+    library: 1,
+    addon: 2,
+    webhost: 3,
+    otherhost: 4,
+    url: 5,
+  };
+}
+
+/**
+ * Export a read-only copy of the resource types.
+ */
+export const resourceTypes = (getResourceTypes(): $Exact<
+  $ReadOnly<$Call<typeof getResourceTypes>>
+>);
+
+export function getEmptyExtensions(): ExtensionTable {
+  return {
+    // Important!
+    // If modifying this structure, please update all callers of this function to ensure
+    // that they are pushing on correctly to the data structure. These pushes may not
+    // be caught by the type system.
+    id: [],
+    name: [],
+    baseURL: [],
+    length: 0,
+  };
+}
+
+export function getDefaultCategories(): CategoryList {
+  return [
+    { name: 'Idle', color: 'transparent' },
+    { name: 'Other', color: 'grey' },
+    { name: 'Layout', color: 'purple' },
+    { name: 'JavaScript', color: 'yellow' },
+    { name: 'GC / CC', color: 'orange' },
+    { name: 'Network', color: 'lightblue' },
+    { name: 'Graphics', color: 'green' },
+    { name: 'DOM', color: 'blue' },
+  ];
+}
+
+export function getEmptyJsTracerTable(): JsTracerTable {
+  return {
+    // Important!
+    // If modifying this structure, please update all callers of this function to ensure
+    // that they are pushing on correctly to the data structure. These pushes may not
+    // be caught by the type system.
+    events: [],
+    timestamps: [],
+    durations: [],
+    lines: [],
+    columns: [],
+    length: 0,
+  };
+}
+
+export function getEmptyThread(overrides?: $Shape<Thread>): Thread {
```

Comment:
maybe write a comment explaining that the argument is meant to be used in tests?

Reply:
I'm fine leaving it open to future consumers using this in app code, even though we currently aren't.